pkg/cmd/roachtest/roachtestutil/mixedversion/mutators_test.go

```diff
@@ -37,18 +37,20 @@ func TestPreserveDowngradeOptionRandomizerMutator(t *testing.T) {
 	require.NotEmpty(t, mutations)
 	require.True(t, len(mutations)%2 == 0, "should produce even number of mutations") // one removal and one insertion per upgrade
 
-	// First half of mutations should be the removals of the existing
-	// `allowUpgradeStep`s.
-	for j := 0; j < numUpgrades/2; j++ {
+	// Mutations should be a removal (of the existing
+	// `allowUpgradeStep`), followed by an insertion of the same step in
+	// a different spot.
+	var j int
+	for j < len(mutations) {
+		// Removal
 		require.Equal(t, mutationRemove, mutations[j].op)
 		require.IsType(t, allowUpgradeStep{}, mutations[j].reference.impl)
-	}
 
-	// Second half of mutations should be insertions of new
-	// `allowUpgradeStep`s.
-	for j := numUpgrades / 2; j < len(mutations); j++ {
-		require.Equal(t, mutationInsertBefore, mutations[j].op)
-		require.IsType(t, allowUpgradeStep{}, mutations[j].impl)
+		// Insertion
+		require.Equal(t, mutationInsertBefore, mutations[j+1].op)
+		require.IsType(t, allowUpgradeStep{}, mutations[j+1].impl)
+
+		j += 2 // check next pair
```

Comment:
Nit: we should update the comment for `mut.Generate` to say that the length of the resulting sequence is always even.

Reply:
That is only true for the `preserveDowngradeOptionRandomizerMutator` implementation I think? Unless you meant just for that specific implementation, in which case I've added it.